pkg/state/release.go

```diff
@@ -178,7 +180,9 @@ func (r ReleaseSpec) ExecuteTemplateExpressions(renderer *tmpl.FileRenderer) (*R
 			}
 		}
 
-		result.SetValues = result.SetValuesTemplate
+		var newvals []SetValue
+		newvals = append(newvals, result.SetValuesTemplate...)
+		result.SetValues = append(newvals, result.SetValues...)
```

Comment:
@jdemeyer Hey! This isn't a blocker, but can I reverse the order of values later?

I think the general usage of `values` in combination with `valuesTemplate` is that, `values` is the default values across environments, and `valuesTemplate`  are used for overlaying specific configuration per environment(or alike). So it should be `Values + ValuesTemplate`, rather than `ValuesTemplate + Values` we have now.

Reply:
I'm not sure that changing the order *later* is such a good idea. Better change it *now*, before people start relying on it.

Ideally, release-specific `values` should come last (after `valuesTemplate`) but default `values` should come first (before `valuesTemplate`). I haven't checked how hard it would be to implement this.

Reply:
Actually, after reading the code, it seems that my change only affects the parsing of `values` and `valuesTemplate` within the `release:` block and it doesn't affect default or environment-specific values. If this analysis is correct, I think that it is doing the right thing.

Reply:
Could you clarify a bit? There's really no "release specic default" values.

Also, environment values has generally nothing to with release specific values.

You can derive release values from env values, but that happens only when you write some go template within helmfile.yaml.

Reply:
"Later" here means a day or two if we could settle on something :) So that we won't break someone's helmfile.yaml.